homeassistant/components/squeezebox/media_player.py

```diff
@@ -169,13 +151,13 @@ async def _discovered_player(player):
             for player in players:
                 hass.async_create_task(_discovered_player(player))
 
-        hass.data[DOMAIN][config_entry.entry_id][PLAYER_DISCOVERY_UNSUB] = (
-            async_call_later(hass, DISCOVERY_INTERVAL, _discovery)
+        entry.runtime_data[PLAYER_DISCOVERY_UNSUB] = (
+            async_call_later(hass, DISCOVERY_INTERVAL, _player_discovery)
         )
```

Comment:
```suggestion
        entry.async_on_unload(
            async_call_later(hass, DISCOVERY_INTERVAL, _player_discovery)
        )
```

Reply:
ah ok nice, I see still get up to speed on @rajlaud code